core/src/main/java/org/hisp/dhis/android/core/program/internal/ProgramModuleDownloader.kt

```diff
@@ -61,53 +61,48 @@ internal class ProgramModuleDownloader @Inject constructor(
+
+        val programUids = getUids(programs)
+        val programStages = programStageCall.download(programUids)
+        val trackedEntityUids = ProgramParentUidsHelper.getAssignedTrackedEntityUids(programs)
+        val trackedEntityTypes = trackedEntityTypeCall.download(trackedEntityUids)
+
+        val attributes = trackedEntityAttributeCall.download(
+            ProgramParentUidsHelper
+                .getAssignedTrackedEntityAttributeUids(programs, trackedEntityTypes),
+        )
+
+
+        val optionSetUids = ProgramParentUidsHelper.getAssignedOptionSetUids(
+            attributes,
+            programStages,
+        )
+        flowOf(
+            programRuleCall.download(programUids),
+            relationshipTypeCall.download(),
+            optionSetCall.download(optionSetUids),
+            optionCall.download(optionSetUids),
+            optionGroupCall.download(optionSetUids),
+
+            ).collect()
+
+        downloadFiltersAndWorkingLists(programUids)
 
-            programCall.download(orgUnitProgramUids)
-                .flatMapCompletable { programs ->
-                    val programUids = getUids(programs)
-                    programStageCall.download(programUids).flatMapCompletable { programStages ->
-                        val trackedEntityUids = ProgramParentUidsHelper.getAssignedTrackedEntityUids(programs)
-                        trackedEntityTypeCall.download(trackedEntityUids)
-                            .flatMap { trackedEntityTypes ->
-                                trackedEntityAttributeCall.download(
-                                    ProgramParentUidsHelper
-                                        .getAssignedTrackedEntityAttributeUids(programs, trackedEntityTypes),
-                                )
-                            }
-                            .flatMapCompletable { attributes ->
-                                val optionSetUids = ProgramParentUidsHelper.getAssignedOptionSetUids(
-                                    attributes,
-                                    programStages,
-                                )
-                                Single.merge(
-                                    listOf(
-                                        programRuleCall.download(programUids),
-                                        relationshipTypeCall.download(),
-                                        optionSetCall.download(optionSetUids),
-                                        optionCall.download(optionSetUids),
-                                        optionGroupCall.download(optionSetUids),
-                                    ),
-                                ).ignoreElements()
-                            }
-                            .concatWith(downloadFiltersAndWorkingLists(programUids))
-                    }
-                }
-        }
     }
 
-    private fun downloadFiltersAndWorkingLists(programUids: Set<String>): Completable {
-        return Single.merge(
```

Comment:
The same here. The point on using `Single.merge` was that it avoids the concatenation with `flatMap`. In this way, the list of calls is flatten. 
Using coroutines, this is not needed anymore because the call can be written in sequence.